### ISSUE TITLE
Don't use defer

### ISSUE THREAD

**Issue by steida** [casetext/fireproof#14]
https://github.com/casetext/fireproof/blob/master/dist/fireproof.js#L153

Since it has been deprecated in Bluebird 3 and therefore it shows unnecessary warnings.

Warning: Promise.defer is deprecated and will be removed in a future version. Use new Promise instead.

**Comment by daguej** [comment]
If your promise library of choice has a deprecated `defer`, remove it before blessing.

```
delete bluebird.defer;
Fireproof.bless(bluebird);
```

**Comment by steida** [comment]
I tried it. Of course it broke something else. Please, remove it. There is no reason to support defer when the code bellow does exactly the same.

**Comment by steida** [comment]
![screen shot 2015-11-20 at 01 10 07](https://cloud.githubusercontent.com/assets/66249/11288341/769b4bfe-8f23-11e5-9d87-c3988095038a.png)

**Comment by daguej** [comment]
Fireproof supports many promise libraries (Q, $q, `Promise`, etc)--not just bluebird--and where `defer` is supplied by the library, we want to use it.

<s>Since Fireproof only checks for the presence of `defer` during `bless()`, you can replace it after:

```
var deprecatedDefer = bluebird.defer;
delete bluebird.defer;
Fireproof.bless(bluebird);
bluebird.defer = deprecatedDefer;
```

</s>

Looks like bluebird might be checking if `defer` is still there?  That's silly.

**Comment by steida** [comment]
Trust me. I tried also this one. And it's ugly workaround also. And defer is not part of specification. And we don't need it anyway.

**Comment by steida** [comment]
There is some check in Bluebird, that's why your suggested hideous workaround doesn't work.

**Comment by daguej** [comment]
We will investigate this issue further, but I'm hesitant to stop using the supplied promise library's built-in `defer` when available.

**Comment by steida** [comment]
The thing is, that method has some nasty consequences as described in Bluebird issues, that's why Bluebird decided to warn users.

**Comment by daguej** [comment]
Resolved by #16

**Comment by steida** [comment]
Thank you guys.
